sweep/sweeper.go

```diff
@@ -1229,6 +1229,12 @@ func (s *UtxoSweeper) getInputLists(cluster inputCluster,
 	// contain inputs that failed before. Therefore we also add sets
 	// consisting of only new inputs to the list, to make sure that new
 	// inputs are given a good, isolated chance of being published.
+	//
+	// TODO(yy): this would lead to conflict transactions as the same input
+	// can be used in two sweeping transactions, and our rebroadcaster will
+	// retry the failed one. We should instead understand why the input is
+	// failed in the first place, and start tracking input states in
+	// sweeper to avoid this.
```

Comment:
Yeah for the full-node backends, in theory we can either try to parse the error message to figure out why the input was rejected, _or_ just re-publish and bisect to see why the transaction was rejected. A third route would be starting to use `tstmempoolaccept`, which I think returns structured information w.r.t why something wouldn't have been accepted.

Reply:
Cool will fix this in the coming PRs which improve our sweeper.